diff --git a/Makefile b/Makefile
index 6ba856c..d0dce0c 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ TOOLCHAIN = $(TOP)/xtensa-lx106-elf
 # for supported versions.
 VENDOR_SDK = 2.0.0
 
-.PHONY: crosstool-NG toolchain libhal libcirom sdk
+.PHONY: toolchain libhal libcirom sdk
 
 
 
@@ -122,29 +122,18 @@ esptool: toolchain
 
 toolchain: $(TOOLCHAIN)/bin/xtensa-lx106-elf-gcc
 
-$(TOOLCHAIN)/bin/xtensa-lx106-elf-gcc: crosstool-NG/ct-ng
-	cp -f 1000-mforce-l32.patch crosstool-NG/local-patches/gcc/4.8.5/
-	make -C crosstool-NG -f ../Makefile _toolchain
+$(TOOLCHAIN)/bin/xtensa-lx106-elf-gcc:
+	mkdir crosstool-NG-config
+	make -C crosstool-NG-config -f ../Makefile _toolchain
 
 _toolchain:
-	./ct-ng xtensa-lx106-elf
+	ct-ng xtensa-lx106-elf
 	sed -r -i.org s%CT_PREFIX_DIR=.*%CT_PREFIX_DIR="$(TOOLCHAIN)"% .config
 	sed -r -i s%CT_INSTALL_DIR_RO=y%"#"CT_INSTALL_DIR_RO=y% .config
 	cat ../crosstool-config-overrides >> .config
-	./ct-ng build
+	ct-ng build
 
 
-crosstool-NG: crosstool-NG/ct-ng
-
-crosstool-NG/ct-ng: crosstool-NG/bootstrap
-	make -C crosstool-NG -f ../Makefile _ct-ng
-
-_ct-ng:
-	./bootstrap
-	./configure --prefix=`pwd`
-	make MAKELEVEL=0
-	make install MAKELEVEL=0
-
 crosstool-NG/bootstrap:
 	@echo "You cloned without --recursive, fetching submodules for you."
 	git submodule update --init --recursive
@@ -163,7 +152,7 @@ $(TOOLCHAIN)/xtensa-lx106-elf/sysroot/usr/lib/libhal.a: $(TOOLCHAIN)/bin/xtensa-
 
 _libhal:
 	autoreconf -i
-	PATH=$(TOOLCHAIN)/bin:$(PATH) ./configure --host=xtensa-lx106-elf --prefix=$(TOOLCHAIN)/xtensa-lx106-elf/sysroot/usr
+	CC='' PATH=$(TOOLCHAIN)/bin:$(PATH) ./configure --host=xtensa-lx106-elf --prefix=$(TOOLCHAIN)/xtensa-lx106-elf/sysroot/usr
 	PATH=$(TOOLCHAIN)/bin:$(PATH) make
 	PATH=$(TOOLCHAIN)/bin:$(PATH) make install
 
